src/app/speechbar.service.spec.ts

```diff
@@ -318,3 +318,22 @@ describe('SpeechbarService.ButtonFacade', () => {
     expect(fb.vocalization).toBe('vocala');
   });
 });
+
+describe('SpeechbarService.utteranceConstruction', () => {
+  it('should append period if no punctuation', inject([SpeechbarService], (service: SpeechbarService) => {
+    const vocals = ['I', 'hello'];
+    expect(service.buildSentence(vocals)).toBe('I hello .');
+  }));
+
+  it('should not append period if punctuation', inject([SpeechbarService], (service: SpeechbarService) => {
+    let vocals = ['I', 'hello?'];
+    expect(service.buildSentence(vocals)).toBe('I hello?');
+    vocals = ['I', 'hello!'];
+    expect(service.buildSentence(vocals)).toBe('I hello!');
```

Comment:
Might be nice to have a test for a period on the end as well?

Reply:
Yes. It would. I don't know why I didn't add it.
I miss parametrized tests.
I'll add it

Reply:
oh man.  Now you need to re-review!